packages/app/components/stackbox/Stackbox.tsx

```diff
@@ -227,20 +228,32 @@ export const Stackbox = () => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [switchNetwork]);
 
-  /**
-   * Form state handler when we select a
-   * strategy
-   */
   useEffect(() => {
     const resetDefaultFormValues = () => {
-      setFromToken(null);
-      setToToken(null);
-      setTokenAmount("");
+      const chainId =
+        chain?.id && !chain?.unsupported ? chain.id : ChainId.GNOSIS;
+
+      deselectStrategy();
+      setFromToken(DEFAULT_TOKENS_BY_CHAIN[chainId].from);
+      setToToken(DEFAULT_TOKENS_BY_CHAIN[chainId].to);
+      setTokenAmount("0.0");
       setFrequency(FREQUENCY_OPTIONS.hour);
       setStartDateTime(new Date(Date.now()));
       setEndDateTime(new Date(endDateByFrequency[frequency]));
     };
```

Comment:
Can't we move this function out of the hook, and then just call it when we need it?

Reply:
I rather have the context to handle the conditional trigger and the hook listening to the condition update, as this condition also updates in the `StrategyCard` component.

Reply:
As we spoke offline, I'll create a new context to handle the Stackbox form values separately and consume that context both in the `StrategyCard` (to reset the form values when needed)  and the `Stackbox` component.

I'll resolve this comment thread for now c: